File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    var answer = 0;
    let eratos = Array.from(new Array(n+1),() => true);
    for (let i=2; i<n+1; i++) {
        if (eratos[i]) {
            for (let j=i*2; j<n+1; j+=i) {
                eratos[j] = false;
            }
        }
    }
    answer = eratos.filter((val)=>val).length-2;
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math

def solution(numbers):
    answer = 0
    result = []
    ln = len(numbers)
    ns = list(map(int,numbers))
    chk = [False]*ln
    st = [['',chk]]
    nst = []
    while st:
        n = st.pop()
        for i in range(ln):
            if not n[1][i]:
                n[1][i] = True
                num = int(str(n[0])+str(ns[i]))
                if pcheck(num) and num not in result:
                    result.append(num)
                nst.append([num,n[1][:]])
                n[1][i] = False
        if not st:
            st = nst
            nst = []
    answer = len(result)
    return answer

def pcheck(n):
    if n==0 or n==1:
        return False
    for i in range(2,int(math.sqrt(n))+1):
        if n%i==0:
            return False
    return True